xblock/reference/plugins.py

```diff
@@ -171,3 +130,42 @@ def __set__(self, xblock, value):
         We don't support this until we figure out what this means
         """
         raise NotImplementedError
+
+#  edX-internal prototype services
+
+
+class FSService(Service):
+    """
+    This is a PROTOTYPE service for storing files in XBlock fields.
+
+    It returns a file system as per:
+    https://github.com/pmitros/django-pyfs
+
+    1) We want to change how load() works, and specifically how
+    prefixes are calculated.
+    2) There is discussion as to whether we want this service at
+    all. Specifically:
+    - It is unclear if XBlocks ought to have filesystem-as-a-service,
+      or just as a field, as per below. Below requires an FS service,
+      but it is not clear XBlocks should know about it.
+    """
+    @public()
+    def load(self, instance, xblock):
+        """
+        Get the filesystem for the field specified in 'instance' and the
+        xblock in 'xblock' It is locally scoped.
+        """
+
+        # TODO: Get xblock from context, once the plumbing is piped through
+        try:
+            return djpyfs.get_filesystem(scope_key(instance, xblock))
+        except NameError:
```

Comment:
The line of code here could raise NameError for a number of reasons.  I'd rather keep the logic at the top of the file that sets djpyfs to None if it can't be imported, and then check it for None here.